enterprise/backend/src/metabase_enterprise/workspaces/types.clj

```diff
@@ -31,3 +31,123 @@
    [:table [:map
             [:name :string]
             [:schema {:optional true} [:maybe :string]]]]])
+
+;;; ---------------------------------------- Graph/Problem Types ----------------------------------------
+
+;; Node types used in workspace graphs and problem detection
+;; Note: :global-transform is for transforms completely outside the workspace (not enclosed, not checked out)
+(mr/def ::node-type [:enum :input-table :external-transform :workspace-transform :global-transform])
```

Comment:
This "global-transform" is a terrible name, because in the glossary `external-transform` is a subset of `global-transform` (which also includes transforms we're going to override).

Perhaps we should repurpose `external` (redefine it) 🤔